wagtail/users/views/users.py

```diff
@@ -70,74 +70,70 @@ def get_users_filter_query(q, model_fields):
 
-    if "q" in request.GET:
-        form = SearchForm(request.GET, placeholder=_("Search users"))
-        if form.is_valid():
-            q = form.cleaned_data["q"]
-            is_searching = True
-            conditions = get_users_filter_query(q, model_fields)
+        if "last_name" in self.model_fields and "first_name" in self.model_fields:
+            users = users.order_by("last_name", "first_name")
 
-            users = User.objects.filter(group_filter & conditions)
-    else:
-        form = SearchForm(placeholder=_("Search users"))
+        if self.get_ordering() == "username":
+            users = users.order_by(User.USERNAME_FIELD)
 
-    if not is_searching:
-        users = User.objects.filter(group_filter)
+        return users
 
-    if "last_name" in model_fields and "first_name" in model_fields:
-        users = users.order_by("last_name", "first_name")
+    @property
+    def requested_with_XMLHttpRequest(self):
+        return self.request.headers.get("x-requested-with") == "XMLHttpRequest"
 
-    if "ordering" in request.GET:
-        ordering = request.GET["ordering"]
+    def get_template(self):
+        if self.requested_with_XMLHttpRequest:
+            return "wagtailusers/users/results.html"
+        else:
+            return "wagtailusers/users/index.html"
 
-        if ordering == "username":
-            users = users.order_by(User.USERNAME_FIELD)
-    else:
-        ordering = "name"
-
-    paginator = Paginator(users.select_related("wagtail_userprofile"), per_page=20)
-    users = paginator.get_page(request.GET.get("p"))
-
-    if request.headers.get("x-requested-with") == "XMLHttpRequest":
-        return TemplateResponse(
-            request,
-            "wagtailusers/users/results.html",
-            {
-                "users": users,
-                "is_searching": is_searching,
-                "query_string": q,
-                "ordering": ordering,
-            },
-        )
-    else:
-        return TemplateResponse(
-            request,
-            "wagtailusers/users/index.html",
-            {
-                "group": group,
-                "search_form": form,
-                "users": users,
-                "is_searching": is_searching,
-                "ordering": ordering,
-                "query_string": q,
-                "app_label": User._meta.app_label,
-                "model_name": User._meta.model_name,
-            },
+    def get_context_data(self, *args, object_list=None, **kwargs):
+        context_data = super().get_context_data(
+            *args, object_list=object_list, **kwargs
         )
+        context_data["ordering"] = self.get_ordering()
+        context_data["group"] = self.group
+        if not self.requested_with_XMLHttpRequest:
+            context_data.update(
+                {
+                    "app_label": User._meta.app_label,
+                    "model_name": User._meta.model_name,
+                }
+            )
+        return context_data
```

Comment:
As above - let's check this request header inline for simplicity, but also let's reverse the logic so that it is easier to read - without the `not` usage.

```suggestion
        if self.request.headers.get("x-requested-with") == "XMLHttpRequest":
            return context_data

        context_data.update(
            {
                "app_label": User._meta.app_label,
                "model_name": User._meta.model_name,
            }
        )
        return context_data
```